### PR TITLE
Generalized AQACF Notebook

### PR DESCRIPTION
**Pull request by sarahkhallam**
Added a generalized AQACF Notebook

### CODE REVIEW

**Comment by grallewellyn** [comment]
I would suggest an environment.yml specifically for the AQ notebook, even though it worked for me using the environment.yml for IDEAS notebook and I was able to successfully run the notebook. However, the last cell (dust storms) doesn't show a gif in the notebook. When I look in my folder, I see MERRA2_Dust_Surface_Mass_Concentration_PM25_Monthly.gif, so I am not sure why it isn't displaying in the notebook. I tried deleting the gif and running the cell again and it still didn't display. Is anyone else experiencing this problem?

Also a note to other reviewers, getting the AQACF data inventory initially used this url: https://digitaltwin.jpl.nasa.gov/nexus/, but that wasn't working so I suggested Sarah change it to https://ideas-digitaltwin.jpl.nasa.gov/nexus/. Is this alright?

Some notes for the contents of the notebook:
- For the compare 2021 and 2022 emissions, I am not sure if we want to display the graph until July 11th if both datasets don't go to July 11th. I think this is something we talked about before when I was working on the notebook. 
test2

- make sure you spell out all acronyms like AMOD and XRF
- Something I wanted to do but never got around to was creating a legend for the generated gif from web map service 
- Does it make sense to have thermal anomaly data in the Air quality notebook? Correct me if I am missing something 
- an interesting feature that IDEAS does a lot is the map with a colormap on top. I only see that for the AQ wildfire use case, but it could be interesting to use it more
test


Next steps for this would be extract the functions from the notebook into python files

**Comment by kevinmarlis** [comment]
A few thoughts:

If we're adding the FireAlarm notebook to this repo we should change a lot of the references to be more EarthDigitalTwin generalized. 

ex: name of the repo, name of the modules directory, name of the processing file in the modules directory, name of the Conda environment in [/IDEAS_notebooks/environment.yml](https://github.com/EarthDigitalTwin/IDEAS-notebooks/blob/main/environment.yml), etc should all lose the `IDEAS` specific naming conventions in favor of something more generalized like EDT.

This means we can reuse what we've already done for setting up the IDEAS notebook (no need to create another `environment.yml` or rewrite all of the abstracted nexus and plotting functions.
___
We also want to make use of those abstracted functions in https://github.com/sarahkhallam/Revised-IDEAS-notebooks/tree/main/ideas_modules. It makes the notebook more user friendly without having to understand a bunch of the code that calls the backends. It also ensures consistency with plot and map styles across both notebooks. The IDEAS notebook shows how to use them - this would entail removing the first few cells that define functions, and replacing the analysis and plotting calls with the `ideas_modules/` functions.
___
I think we want to highlight more than just the time series algorithm (the `spatial_mean` function) and the variance algorithm (the `temporal_variance` function). Some other options are the daily differences algorithm using tropomi climatology for la ports example, time average map algorithm (`temporal_mean` function) which is good for mapping the average values over a given area, and the max min map algorithm which is good for mapping the maximum values over a given area.
___
@jtroberts I'd love your thoughts on this.

**Comment by jtroberts** [comment]
I agree with @kevinmarlis 's comments, but not sure about losing the IDEAS specific naming conventions. "IDEAS" is already a generic name and matches our server name.

**Comment by jtroberts** [comment]
@sarahkhallam can you push your recent updates for review?

**Comment by sarahkhallam** [comment]
Just pushed

On Apr 13, 2023, at 10:43 AM, Joe T. Roberts ***@***.***> wrote:



@sarahkhallam<https://urldefense.us/v3/__https://github.com/sarahkhallam__;!!PvBDto6Hs4WbVuu7!J1FCp0bwUfxnN0Ep6yWP8uX2nrBG9MaKHk7b1R3Hp0EkDr_87QjpgeHIgzLGVdOejRChKsIapoM05BBPpLHkojEmkckg8KmreA$> can you push your recent updates for review?

File: Generalized_AQACF.ipynb
Comment (kevinmarlis) [inline Generalized_AQACF.ipynb:1]: The GMU data can be accessed through the `ideas-digital twin` URL, meaning the cell that defines `gmu_base_url` is not needed. And the example hitting the GMU server for the NO2 dataset should be modified. Right now it is hitting the GMU server for a dataset that they are proxying to us for. Instead, we should be showing a case where we proxy to one of their datasets. The `yellow_sea_omi_no2` cell is an example of that. 

Their datasets that we support via proxy are included in the nexus/list endpoint and are setup like `GMU-{collection_name}`

File: ideas_modules/plotting.py
Comment (kevinmarlis) [inline ideas_modules/plotting.py:1]: The functions added here already exist in `ideas_modules/processing.py`. Can the notebook use those instead as this script should only contain functions used for plotting.

Comment (kevinmarlis) [reply ideas_modules/plotting.py:1]: I edited the AQ notebook to use the shared processing and plotting functions. I made one content edit to remove the O3 variance example as the data at the source has some dropouts during that period (hence that weird looking map).

**Comment by kevinmarlis** [comment]
I think this is looking good now, although I don't know if there needs to be more content edits. Hopefully there won't be very many merge conflicts because I was also making some edits on the IDEAS side.

**Pull request merged**